packages/@aws-cdk/aws-ecs/lib/base/base-service.ts

```diff
@@ -546,6 +587,11 @@ export abstract class BaseService extends Resource
           enable: true,
           rollback: props.circuitBreaker.rollback ?? false,
         } : undefined,
+        alarms: props.deploymentAlarms ? {
+          alarmNames: props.deploymentAlarms.alarms.map(alarm => alarm.alarmName),
+          enable: true,
+          rollback: props.deploymentAlarms.behavior ? props.deploymentAlarms.behavior !== AlarmBehavior.FAIL_ON_ALARM : true,
```

Comment:
when we try to use positive instead of negative it giving an error that 

Whether to enable the deployment alarms. If this property is defined, alarms will be implicitly enabled.
@default
- disabled
This comparison appears to be unintentional because the types 'AlarmBehavior.FAIL_ON_ALARM' and 'AlarmBehavior.ROLLBACK_ON_ALARM' have no overlap.

So I'm rewriting this as ` rollback: props.deploymentAlarms.behavior !== AlarmBehavior.FAIL_ON_ALARM,`